contracts/root/RootChain.sol

```diff
@@ -22,50 +24,97 @@ contract RootChain is RootChainStorage, IRootChain {
+    //     extraData = extraData[0].toList();
+    //     RootChainHeader.HeaderBlock memory headerBlock = _buildHeaderBlock(
+    //         extraData
+    //     );
+    //     headerBlocks[_nextHeaderBlock] = headerBlock;
+    //     // check if it is better to keep it in local storage instead
+    //     IStakeManager stakeManager = IStakeManager(
+    //         registry.getStakeManagerAddress()
+    //     );
+    //     // blockInterval, voteHash, stateRoot, sigs
+    //     uint256 _reward = stakeManager.checkSignatures(
+    //         headerBlock.end.sub(headerBlock.start).add(1),
+    //         keccak256(vote),
+    //         bytes32(extraData[4].toUint()),
+    //         sigs
+    //     );
+    //     require(_reward != 0, "Invalid checkpoint");
+    //     emit NewHeaderBlock(
+    //         headerBlock.proposer,
+    //         _nextHeaderBlock,
+    //         _reward,
+    //         headerBlock.start,
+    //         headerBlock.end,
+    //         headerBlock.root
+    //     );
+    //     _nextHeaderBlock = _nextHeaderBlock.add(MAX_DEPOSITS);
+    //     _blockDepositId = 1;
+    // }
+
+    // side channel
+    function submitHeaderBlock(bytes calldata data, bytes calldata sigs)
+        external
+    {
+        (
+            address proposer,
+            uint256 start,
+            uint256 end,
+            bytes32 rootHash,
+            bytes32 accountHash,
+            bytes32 _borChainID
+        ) = abi.decode(
+            data,
+            (address, uint256, uint256, bytes32, bytes32, bytes32)
         );
-        require(dataList[1].toUint() == VOTE_TYPE, "Vote type is invalid");
+        // require(bytes32(uint(15001)) == _borChainID, "Invalid bor chain id");
 
-        // validate hash of txData was signed as part of the vote
         require(
-            keccak256(dataList[4].toBytes()) ==
-                keccak256(abi.encodePacked(sha256(txData))),
-            "Extra data is invalid"
+            _buildHeaderBlock(proposer, start, end, rootHash),
+            "INCORRECT_HEADER_DATA"
         );
 
-        RLPReader.RLPItem[] memory extraData = txData.toRlpItem().toList();
-        extraData = extraData[0].toList();
-        RootChainHeader.HeaderBlock memory headerBlock = _buildHeaderBlock(
-            extraData
-        );
-        headerBlocks[_nextHeaderBlock] = headerBlock;
         // check if it is better to keep it in local storage instead
         IStakeManager stakeManager = IStakeManager(
             registry.getStakeManagerAddress()
         );
+
         // blockInterval, voteHash, stateRoot, sigs
         uint256 _reward = stakeManager.checkSignatures(
-            headerBlock.end.sub(headerBlock.start).add(1),
-            keccak256(vote),
-            bytes32(extraData[4].toUint()),
+            end.sub(start).add(1),
+            keccak256(abi.encodePacked(bytes(hex"01"), data)), // prefix 01 to data
+            accountHash,
```

Comment:
@0xAshish This should be root, not accountHash

Reply:
seems correct to me